homeassistant/components/mqtt/discovery.py

```diff
@@ -65,6 +67,16 @@ def async_device_message_received(topic, payload, qos):
             payload[CONF_STATE_TOPIC] = '{}/{}/{}/state'.format(
                 discovery_topic, component, object_id)
 
+        discovery_hash = json.dumps([component, object_id], sort_keys=True)
```

Comment:
I think we need to store a unique pair "component - object_id". I don't know can we use a dictionary for this purpose.
Unfortunately, I am not a Python programmer. And I don't have enough experience for write and check tests. I'll try but a result is unknown.